Automation/LettersDispatcherTest.cs

```diff
@@ -28,102 +28,145 @@ public LettersDispatcherTest()
 
-            this.ageValidator.Verify(
-                x => x.IsNotOlderEnough(
-                    It.Is<Person>(y => y.Equals(personOlderEnough))),
-                Times.Once);
+            MinorSender();
 
-            this.badWordsValidator.Verify(
-                x => x.ThereAreNotBadWords(
-                    It.Is<string>(y => y == letterShouldBeSend.Body)),
-                Times.Once);
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
 
-            receiver.ReceivedLetters.Should().BeEquivalentTo(expectedLettersSent);
+            // Assert
+            receiver.ReceivedLetters.Should().HaveCount(0);
         }
 
         [Fact]
         public void Should_NotifyToTheSenderFamily_When_SenderIsNotOlderEnough()
         {
+            // Arrange 
+            var receiver = new Person();
+
+            var momName = "Liz";
+
+            var mom = new Person{ Name=momName };
+
+            var family = new List<Person>{mom};
+
+            var letters = new List<Letter> { new Letter { Title="A message", Sender = new Person { Relatives = family }, Receivers = new List<Person>{ receiver }, Body = "Some Message" } }; 
+
+            MinorSender();
+
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
+
+            // Assert
+            // NOTE: This is kind of a slippery slope. This test uses reference equality since Person doesn't implement Equals.
+            notificationSender.Verify(n => n.Send(It.IsAny<string>(), family), Times.AtLeastOnce());
         }
 
         [Fact]
         public void Should_NotSendLetter_When_TheBodyOfItHasBadWords()
         {
+            // Arrange 
+            var receiver = new Person();
+
+            var letters = new List<Letter> { new Letter { Title="A message", Sender = new Person {}, Receivers = new List<Person>{ receiver }, Body = "Some message" } }; 
+
+            AdultSender();
+
+            BadWords();
+
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
+
+            // Assert
+            receiver.ReceivedLetters.Should().HaveCount(0);
         }
 
         [Fact]
         public void Should_SendLetterToReceivers_When_AllValidationsPass()
         {
+            // Arrange 
+            var receiver = new Person();
+
+            var letters = new List<Letter> { new Letter { Title="A message", Sender = new Person {}, Receivers = new List<Person>{ receiver }, Body = "Some Message"  } }; 
+
+            AdultSender();
+
+            NoBadWords();
+
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
+
+            // Assert
+            receiver.ReceivedLetters.Should().HaveCount(1); // Input drives output. No need for exact equality test.
+        }
+
+        [Fact]
+        public void Should_SendLetterToReceivers_When_ThereIsNoMessage()
```

Comment:
change to Should_NotSendLetterToReceivers_When_ThereIsNoMessage

Reply:
Naming fixed.